packages/opentelemetry/src/utils/spanTypes.ts

```diff
@@ -22,7 +22,7 @@ export function spanHasAttributes<SpanType extends AbstractSpan>(
  */
 export function spanHasKind<SpanType extends AbstractSpan>(span: SpanType): span is SpanType & { kind: SpanKind } {
   const castSpan = span as ReadableSpan;
-  return !!castSpan.kind;
+  return typeof castSpan.kind === 'number';
```

Comment:
m: Quick explainer for this?

Reply:
sorry, had a comment here earlier but changed it again - this check was actually wrong, because this can be `0` which is valid, but we interpreted this as `no kind` before.